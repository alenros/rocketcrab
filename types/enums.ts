export enum PartyStatus {
    loading = "loading",
    party = "party",
    ingame = "ingame",
}

export enum GameStatus {
    loading = "loading",
    inprogress = "inprogress",
    waitingforhost = "waitingforhost",
    error = "error",
}

export enum SocketEvent {
    CONNECT = "connect",
    DISCONNECT = "disconnect",
    RECONNECT = "reconnect",
    GAME_EXIT = "GAME_EXIT",
    GAME_SELECT = "GAME_SELECT",
    GAME_START = "GAME_START",
    HOST_GAME_LOADED = "HOST_GAME_LOADED",
    INVALID_NAME = "INVALID_NAME",
    INVALID_PARTY = "INVALID_PARTY",
    JOIN_PARTY = "JOIN_PARTY",
    NAME = "NAME",
    UPDATE = "UPDATE",
    FINDER_SUBSCRIBE = "FINDER_SUBSCRIBE",
    FINDER_UPDATE = "FINDER_UPDATE",
    CHAT_MESSAGE = "CHAT_MESSAGE",
    KICK_PLAYER = "KICK_PLAYER",
    SET_IS_PUBLIC = "SET_IS_PUBLIC",
}

export enum RocketcrabMode {
    MAIN, // will be shown on rocketcrab.com
    KIDS, // will be shown on kids.rocketcrab.com
    ALL, //  do not use. for testing only.
}
