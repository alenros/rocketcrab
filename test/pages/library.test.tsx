import React from "react";
import { render } from "../testUtils";
import Library from "../../pages/library";
import { ClientGameLibrary } from "../../types/types";
import { RocketcrabMode } from "../../types/enums";

describe("pages/library.tsx", () => {
    it("matches snapshot", () => {
        const { asFragment } = render(
            <Library
                gameLibrary={{} as ClientGameLibrary}
                mode={RocketcrabMode.MAIN}
            />,
            {}
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
