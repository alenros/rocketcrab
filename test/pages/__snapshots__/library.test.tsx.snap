// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/library.tsx matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="jsx-1855009027 container"
  >
    <main
      class="jsx-1855009027"
    >
      <div
        class="jsx-1855009027 main-content"
      >
        <div
          class="jsx-1758552329 title"
        >
          <div
            class="jsx-1758552329"
            style="margin: .5em;"
          >
            <img
              class="jsx-1758552329 rocket"
              src="/rocket.svg"
            />
            <img
              class="jsx-1758552329 crab"
              src="/crab.svg"
            />
          </div>
          <div>
            <div
              style="display: inline-block; white-space: nowrap;"
            >
              <div
                class="jsx-418573366 tooltip "
              >
                <h2
                  class="jsx-1758552329 party-url"
                >
                  rocketcrab.com
                </h2>
              </div>
            </div>
          </div>
        </div>
        <div
          style="text-align: center; justify-content: center;"
        >
          <span
            class="jsx-3499487811 "
          />
          <h4>
            Games
          </h4>
          <span
            class="jsx-870436695 "
          />
          <div
            class="jsx-984614781 with-label"
          >
            <div
              class="jsx-984614781 input-container "
            >
              <div
                class="jsx-984614781 input-wrapper   "
              >
                <span
                  class="jsx-1683013590 input-icon"
                >
                  üîé
                </span>
                <input
                  autocomplete="off"
                  class="jsx-984614781  left-icon"
                  placeholder="Search"
                  type="text"
                  value=""
                />
                <div
                  class="jsx-1717893035 clear-icon "
                >
                  <svg
                    class="jsx-1717893035"
                    fill="none"
                    shape-rendering="geometricPrecision"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                  >
                    <path
                      class="jsx-1717893035"
                      d="M18 6L6 18"
                    />
                    <path
                      class="jsx-1717893035"
                      d="M6 6l12 12"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <span
            class="jsx-870436695 "
          />
          <div
            class="jsx-974242038 item  jsx-1720607024 "
          />
          <span
            class="jsx-870436695 "
          />
          <div
            class="jsx-1938348541 item justify jsx-1720607024 "
          >
            <div
              class="jsx-974242038 item  jsx-4216129763 "
            >
              <button
                class="jsx-1244829081 btn "
                style="width: auto;"
                type="button"
              >
                <div
                  class="text"
                >
                  ‚Ü©Ô∏è Back to rocketcrab.com
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="jsx-1855009027 height-expander"
      />
    </main>
    <footer
      class="jsx-3128977650"
      style="margin-top: 1em;"
    >
      rocketcrab by 
      <a
        class="jsx-3128977650"
        href="https://www.tannerkrewson.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        Tanner Krewson
      </a>
      <br
        class="jsx-3128977650"
      />
      <a
        class="jsx-3128977650"
        href="https://github.com/tannerkrewson/rocketcrab"
        rel="noopener noreferrer"
        target="_blank"
      >
        View on GitHub
      </a>
    </footer>
  </div>
</DocumentFragment>
`;
